views/pages/AdminPage/index.jsx

```diff
@@ -0,0 +1,201 @@
+import React, { useEffect, useState } from 'react'
+import { useOutletContext } from 'react-router-dom'
+import { Box, Button, Modal } from '@mui/material'
+
+import { UsersModel, StudiesModel } from '../../models'
+import api from '../../api'
+import PageHeader from '../../components/PageHeader'
+import AdminUsersTable from '../../tables/AdminUsersTable'
+import { AdminUsersSearchForm } from '../../forms/AdminUsersSearchForm'
+import { MultiSelect } from '../../forms/MultiSelect'
+
+const modalStyle = {
+  position: 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: 720,
+  bgcolor: 'background.paper',
+  border: '2px solid #000',
+  boxShadow: 24,
+  p: 4,
+};
+
+const AdminPage = () => {
+  const usersKey = 'users'
+  const { user, setNotification, users, setUsers } = useOutletContext()
+
+  const [searchOptions, setSearchOptions] = useState([])
+  const [studies, setStudies] = useState([])
+  const [searchOptionsValue, setSearchOptionsValue] = useState([])
+  const [currentUserAccess, setCurrentUserAccess] = useState([])
+  const [openResetKeyModal, setOpenResetKeyModal] = useState(false)
+  const [resetKeyValue, setResetKeyValue] = useState('')
+  const [currentRowIndex, setCurrentRowIndex] = useState(null)
+  const [openAccessModal, setOpenAccessModal] = useState(false)
+
+  const filteredUsers = (searchOptionsValue || []).length ? users.filter(u => searchOptionsValue.map(s => s.value).includes(u.uid)) : users
```

Comment:
```suggestion
  const filteredUsers = (searchOptionsValue || []).length ? users.filter(currentUser => searchOptionsValue.map(searchOption => searchOption.value).includes(currentUser.uid)) : users
```
Nit but the parameter names can be descriptive

Reply:
0a8e6ad